gtk3/sample/misc/tooltips.rb

```diff
@@ -5,28 +5,27 @@
 
   Copyright (c) 2007-2015 Ruby-GNOME2 Project Team
   This program is licenced under the same licence as Ruby-GNOME2.
-
-  $Id: tooltips.rb,v 1.1 2007/07/10 13:17:34 ggc Exp $
 =end
 
-require 'gtk3'
+require "gtk3"
 
-unless Gtk::Version.or_later?(2, 12, 0)
-    puts "This sample requires GTK+ 2.12.0 or later"
-    exit
+unless Gtk::Version.or_later?(3, 4, 2)
+  puts "This sample requires GTK+ 3.4.2 or later: #{Gtk::Version::STRING}"
+  exit
 end
 
+
 def treeview_query_tooltip(treeview, keyboard_tip, x, y, tooltip)
     if keyboard_tip
         # Keyboard mode
-        path, = treeview.cursor
+        ok,path = treeview.cursor
         if not path
             return false
         end
     else
         bin_x, bin_y = treeview.convert_widget_to_bin_window_coords(x, y)
         # Mouse mode
-        path, = treeview.get_path_at_pos(bin_x, bin_y)
+        ok, path = treeview.get_path_at_pos(bin_x, bin_y)
```

Comment:
Yes for the C api but  I changed because if I let the initial line, I had an error in line 34 saying invalid argument TrueClass (expect Gtk::TreePath). The path variable was a boolean. If I print it I have an array with : [true, #<Gtk::TreePath:0x2b3dd30 ptr=0x3c7ef20 own=true>, #<Gtk::TreeViewColumn:0x2e4d1d8 ptr=0x3d27d90>, 319, 0]